ansible/playbooks/adhoc/hotfix/roles/update_master/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+
+# Default to False before attempting upgrade
+- set_fact: 
+    update_successful:    False
+    downgrade_successful: False
+  when: "{{ update_needed == True }}" 
+
+- name: Update atomic-openshift-master
+  yum:
+    name: "{{ rpm_wanted }}"
+    state: present
+  when: "{{ update_needed == True }}" 
+
+- name: Check if update is successful
+  command: "rpm -qa atomic-openshift"
+  register: post_upgrade
+  changed_when: False
+
+# ansible will return a '17' if a match is found. '-1' means 'not found'
+- name: Set update_successful=True if upgrade succeeded
+  set_fact: 
+    update_successful: True
+  when: "{{ post_upgrade.stdout.find(version_wanted) != -1 }}"
+
+- name: Upgrade failed, so try a downgrade
+  command: "yum -y downgrade atomic-openshift-{{ version_wanted }} atomic-openshift-clients-{{ version_wanted }} atomic-openshift-master-{{ version_wanted }} atomic-openshift-node-{{ version_wanted }} tuned-profiles-atomic-openshift-node-{{ version_wanted }} atomic-openshift-sdn-ovs-{{ version_wanted }}"
+  when: "{{ update_successful == False }}"
```

Comment:
@jgouldin we can, yes, but it would still be up to the user to write down that version for future reference. (Since we want to be able to roll back days after the upgrade). 

There is actually a debug statement that says what the currently running version is for all the hosts. It says something like "version installed: ****, version wanted: *****, upgrade needed? True/False". So that output is there for the user if they want to read it as part of the script run.